src/main/java/org/opensearch/ad/transport/StatsAnomalyDetectorTransportAction.java

```diff
@@ -115,33 +120,33 @@ private ActionListener<ADStatsResponse> getRestStatsListener(ActionListener<Stat
      * Make async request to get the number of detectors in AnomalyDetector.ANOMALY_DETECTORS_INDEX if necessary
      * and, onResponse, gather the cluster statistics
      *
-     * @param client Client
+     * @param sdkRestClient SDKRestClient
      * @param listener MultiResponsesDelegateActionListener to be used once both requests complete
      * @param adStatsRequest Request containing stats to be retrieved
      */
     private void getClusterStats(
-        Client client,
+        SDKRestClient sdkRestClient,
         MultiResponsesDelegateActionListener<ADStatsResponse> listener,
         ADStatsRequest adStatsRequest
     ) {
         ADStatsResponse adStatsResponse = new ADStatsResponse();
         if ((adStatsRequest.getStatsToBeRetrieved().contains(StatNames.DETECTOR_COUNT.getName())
             || adStatsRequest.getStatsToBeRetrieved().contains(StatNames.SINGLE_ENTITY_DETECTOR_COUNT.getName())
             || adStatsRequest.getStatsToBeRetrieved().contains(StatNames.MULTI_ENTITY_DETECTOR_COUNT.getName()))
-            && clusterService.state().getRoutingTable().hasIndex(AnomalyDetector.ANOMALY_DETECTORS_INDEX)) {
+            && sdkClusterService.state().getRoutingTable().hasIndex(AnomalyDetector.ANOMALY_DETECTORS_INDEX)) {
 
             TermsAggregationBuilder termsAgg = AggregationBuilders.terms(DETECTOR_TYPE_AGG).field(AnomalyDetector.DETECTOR_TYPE_FIELD);
             SearchRequest request = new SearchRequest()
                 .indices(AnomalyDetector.ANOMALY_DETECTORS_INDEX)
                 .source(new SearchSourceBuilder().aggregation(termsAgg).size(0).trackTotalHits(true));
 
-            client.search(request, ActionListener.wrap(r -> {
-                StringTerms aggregation = r.getAggregations().get(DETECTOR_TYPE_AGG);
-                List<StringTerms.Bucket> buckets = aggregation.getBuckets();
+            sdkRestClient.search(request, ActionListener.wrap(r -> {
+                ParsedStringTerms aggregation = r.getAggregations().get(DETECTOR_TYPE_AGG);
```

Comment:
Why are we changing the type here from `StringTerms` to `ParsedStringTerms`?

Reply:
I am implementing this change due to type cast error. As we are using SDKRestClient so the aggregation is now returning ParsedStringTerms instead of StringTerms. So I have to make this change.

Reply:
Yep. As discussed we will be going in with this change. Can you add an entry in plugin migration doc to address this change? You can put it under a new title "Change data types for RestClient"

Reply:
Opened the PR to address the above: https://github.com/opensearch-project/opensearch-sdk-java/pull/673